adapters/tlsio_openssl.c

```diff
@@ -850,8 +852,12 @@ static int load_cert_crl_http(
         goto error;
     }
 
-    bio = BIO_new_connect(host);
-    if (!bio || !BIO_set_conn_port(bio, port))
+    const char* proxyHostnamePort;
+    const char* usernamePassword;
+    platform_get_http_proxy(&proxyHostnamePort, &usernamePassword);
+
+    bio = BIO_new_connect((proxyHostnamePort && *proxyHostnamePort) ? proxyHostnamePort : host);
```

Comment:
confusing. why are we passing port OR host? these are completely different things like int vs string? does it mean bio_new_connect can handle both? or our names are totally obfuscated?

Reply:
bio uses both, it gets a string which may contain a "host:port"